src/x/parsers/metric_parsers.go

```diff
@@ -0,0 +1,34 @@
+package parsers
+
+import (
+	"github.com/m3db/m3/src/metrics/metric/id"
+	"github.com/m3db/m3/src/x/serialize"
+	"bytes"
+)
+
+func getMetricTagsIterator() serialize.MetricTagsIterator {
+	it := serialize.NewUncheckedMetricTagsIterator(serialize.NewTagSerializationLimits())
```

Comment:
is it possible we instantiate this once in aggregator.go to reduce some cpu cycles for instantiation and gc?

Reply:
the metricIDs can be iterated in parallel, so then we'd need to lock this iterator :/ might be more complicated change

Reply:
what if we monitor CPU% in dev deployment and optimize if needed?